stats.py

```diff
@@ -43,29 +41,17 @@ def retention_stability(lim) -> tuple:
     )
     # x[0]: elapsed days
     # x[1]: stability
-    # x[2]: same nid count
-    # x[3]: nid
     elapse_stability_list = filter(lambda x: x[1] is not None, elapse_stability_list)
```

Comment:
```suggestion
```

I don't think that this line is required because we already have `AND json_extract(data, '$.s') IS NOT NULL` in the SQL.